webapps/frontend/app/viewmodels/site/backend-alba-detail.js

```diff
@@ -297,31 +297,42 @@ define([
             }
         };
         self.link = function() {
-            var diskNames = [];
+            var diskNames = [], nodeDisks = {};
+            $.each(self.registeredNodes(), function(index, node) {
+                nodeDisks[node.nodeID()] = {
+                    node: node,
+                    disks: node.disks(),
+                    names: node.diskNames(),
+                    changed: false
+                };
+            });
             $.each(self.disks(), function (index, disk) {
                 diskNames.push(disk.name());
                 if (disk.node === undefined) {
-                    $.each(self.registeredNodes(), function(jndex, node) {
-                        if (disk.nodeID() === node.nodeID()) {
-                            disk.node = node;
-                            node.disks.push(disk);
-                            node.disks.sort(function (a, b) {
-                                return a.name() < b.name() ? -1 : 1;
-                            });
+                    $.each(nodeDisks, function(nodeID, nodeInfo) {
+                        if (disk.nodeID() === nodeID && !nodeInfo.names.contains(disk.name())) {
+                            disk.node = nodeInfo.node;
+                            nodeInfo.disks.push(disk);
```

Comment:
It shouldn't be a problem if there are no duplicate disks (which won't be the case since their name is primary key, and they are loaded at line 301), but it's nicer anyway.